src/Microsoft.Azure.WebJobs.Host/Extensions/ExceptionExtensions.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Threading;
+using Microsoft.Azure.WebJobs.Host;
+
+namespace System
+{
+    internal static class ExceptionExtensions
+    {
+        public static bool IsFatal(this Exception exception)
+        {
+            while (exception != null)
+            {
+                if ((exception as OutOfMemoryException != null && exception as InsufficientMemoryException == null) ||
```

Comment:
The first check/cast is not needed. With the current logic, you'd only need to check for `InsufficientMemoryException`. I'm assuming what we really want is to interpret `OutOfMemoryException` as fatal, in which case, the second check is unnecessary.